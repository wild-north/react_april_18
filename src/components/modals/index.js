import React, { Fragment} from 'react';
import { Modal, Button } from 'react-bootstrap';
import { modalConnector } from './connector';

export const ModalComponent = (props) => {
    const { title, show, children, hideModal, onHide, buttons } = props;

    return (
        <Modal show={ show } onHide={ onHide }>
            <Modal.Header closeButton>
            {
                title && (
                    <Modal.Title>{ title }</Modal.Title>
                )
            }
            </Modal.Header>
            <Modal.Body>
                { children }
            </Modal.Body>
            <Modal.Footer>
                {
                    buttons ? buttons : (
                        <div className="text-center">
                            <Button onClick={ onHide }>Close</Button>
                        </div>
                    )
                }
            </Modal.Footer>
        </Modal>
    );
};

export default modalConnector(ModalComponent);