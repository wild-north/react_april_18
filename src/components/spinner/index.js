import React from 'react';

export const Spinner = () => (
    <img src="loading_spinner.gif" alt="spinner"/>
);