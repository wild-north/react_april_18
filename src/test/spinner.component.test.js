import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { SpinnerForTest } from '../components/spinner';

it('renders correctly', () => {
    // const tree = renderer
    //     .create(<SpinnerForTest/>)
    //     .toJSON();
    const wrapper = shallow(<SpinnerForTest/>);
    expect(wrapper).toMatchSnapshot();
});