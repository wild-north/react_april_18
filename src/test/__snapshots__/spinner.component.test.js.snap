// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="spinner"
>
  <span
    className="spinner-content"
  >
    <img
      alt="spinner"
      src="loading_spinner.gif"
    />
  </span>
</div>
`;
